### ISSUE TITLE
Nested Serialization throwing error for nil attributes 

### ISSUE THREAD

**Issue by suchitpuri** [rails-api/active_model_serializers#490]
Hi , 

We have a structure of models similar to the below code 

```ruby

class DemoService::OrganizationSerializer < ActiveModel::Serializer
	attributes :id , :name, 
end

class DemoService::ContactSerializer < ActiveModel::Serializer
	attributes :id , :first_name, :surname, 
	has_one :organization , serializer: DemoService::OrganizationSerializer
	embed :ids , include: true
end

class DemoService::ClientSerializer < ActiveModel::Serializer
	attributes :amount , :id 
	has_many :references
	has_one :instructing_contact ,serializer: DemoService::ContactSerializer 
	has_one :billing_contact , serializer: DemoService::ContactSerializer 
	embed :ids , include: true
end

class DemoService::JobDetailSerializer < ActiveModel::Serializer
	attributes :send_report_by,:service_type,:comment,:turnaround_days,:turnaround_comment,:enquiry_id,:enquiry_source,:hard_copy_reports,:referral_name
end

class DemoService::EnquirySerializer < ActiveModel::Serializer
  attributes :number , :id 
  has_one :job_detail,serializer: DemoService::JobDetailSerializer
  has_many :clients , serializer: DemoService::ClientSerializer
  has_many :property_metadata
  embed :ids , include: true
end

```
So if the billing_contact is null , its not able to build any serializer and send an error that "cannot find organization for nil class"

so in order to fix it we had to do 

```ruby
class ActiveModel::Serializer::Association
	def build_serializer(object, options = {})
		return ActiveModel::DefaultSerializer.new(object, options.merge(@options)) unless object
		@serializer_class.new(object, options.merge(@options))
	end
end

```

Please let me know if this solution is fine so that i can send a pull request.

Thanks

**Comment by kurko** [comment]
Can you try again now from the `0-9-stable` branch? #797 was merged and likely fixed this. Please reopen if it doesn't work.

**Comment by jpaas** [comment]
I'm still seeing this on 0.9.3

**Comment by afn** [comment]
Thanks @r4m! Can you push a new version to rubygems please?

**Comment by joaomdmoura** [comment]
@afn We have a new release schedule for tomorrow actually

**Comment by cesarjr** [comment]
I'm having the same trouble using has_one at 0.9.3.

**Comment by linuxlewis** [comment]
I'm having this issue on 0.9.3 as well

**Comment by linuxlewis** [comment]
Seems rubygems is just out of date. Installing from the 0-9-stable branch fixed for me :+1:

**Comment by bf4** [comment]
@kurko @joaomdmoura seems we should release a `0.9.4`  [https://github.com/rails-api/active_model_serializers/compare/v0.9.3...0-9-stable](https://github.com/rails-api/active_model_serializers/compare/v0.9.3...dc08f0e) ?

**Comment by jay-clark** [comment]
Is this release going to happen? Just hit this issue on has_one also.

**Comment by cesarjr** [comment]
+1

**Comment by bf4** [comment]
@wycats @steveklabnik @joaomdmoura @josevalim @spastorino @kurko would love for someone with push rubygems privileges to cut 0.9.4...

**Comment by spastorino** [comment]
@bf4 I'm not actually maintaining AMS. I think @kurko and/or @joaomdmoura are the ones, now.

**Comment by bf4** [comment]
Yeah, but you have gem ownership and can push :)

B mobile phone

**Comment by josevalim** [comment]
Please ask one of the maintainers to contact us if they need to push permission. It would be rude of us to just publish a project we no longer maintain.

**Comment by josevalim** [comment]
@bf4 that clarifies *a lot*! :D sorry for the confusion! Can you please let me know your rubygems e-mail? you can spell it out here or send me an e-mail (you can find my e-mail on my profile).

**Comment by josevalim** [comment]
I have added you and removed myself. :)

**Comment by joaomdmoura** [comment]
Just read it, glad you figure it out, let me know in case you still need any help @bf4, was offline this xmas + new year

**Comment by bf4** [comment]
Released!  0.9.4 https://twitter.com/rubygems/status/684456579471683586 https://rubygems.org/gems/active_model_serializers/versions/0.9.4

I tried to [update the change log but the pr history across branches was a little weird. Lemme know](https://github.com/rails-api/active_model_serializers/commits/0-9-stable)

**Comment by afn** [comment]
:beer: Thanks!
